hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/contracts/execution/traceability/HederaTracer.java

```diff
@@ -73,6 +83,27 @@ public void init(final MessageFrame initialFrame) {
         }
     }
 
+    @Override
+    public void finalizeOperation(final MessageFrame initialFrame) {
+        if (areActionSidecarsEnabled) {
+            // make sure only valid actions are left in allActions
+            final var validActions =
+                    allActions.stream().filter(SolidityAction::isValid).toList();
+
+            if (!currentActionsStack.isEmpty() || validActions.size() != allActions.size()) {
+                log.atDebug().log(
+                        "HederaTracer invalid at end of EVM run: {} ({})",
+                        () -> formatAnomaliesAtFinalizationForLog(validActions),
```

Comment:
Should anomalies be at debug?  This to me indicates we got the logic wrong in potentially unsafe ways.  I expect they should never happen, so an anomoly should be logged at least at info.

Reply:
let's talk offline about these logs - I may have misunderstood the latest instructions